// @aws-sdk/client-lambda/dist-es/commands/UpdateFunctionEventInvokeConfigCommand.js
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { FunctionEventInvokeConfigFilterSensitiveLog, UpdateFunctionEventInvokeConfigRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateFunctionEventInvokeConfigCommand, serializeAws_restJson1UpdateFunctionEventInvokeConfigCommand, } from "../protocols/Aws_restJson1";
export class UpdateFunctionEventInvokeConfigCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "LambdaClient";
        const commandName = "UpdateFunctionEventInvokeConfigCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: UpdateFunctionEventInvokeConfigRequestFilterSensitiveLog,
            outputFilterSensitiveLog: FunctionEventInvokeConfigFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restJson1UpdateFunctionEventInvokeConfigCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restJson1UpdateFunctionEventInvokeConfigCommand(output, context);
    }
}
